### ISSUE TITLE
It should be "south_atlantic" instead of "south_america" or "australia"  for basin?

### ISSUE THREAD

**Issue by GC-OCEAN** [tropycal/tropycal#164]
I am little confusing with the API reference. It said that, basin can be set to "south_america" or "australia" for IBTrACS source. But there are no such basins for IBTrACS.

If you run `basin = tracks.TrackDataset(basin='south_america',source='ibtracs')`, you will get

---------------------------------------------------------------------------
HTTPError                                 Traceback (most recent call last)
Input In [25], in <cell line: 1>()
----> 1 basin = tracks.TrackDataset(basin='south_america',source='ibtracs')

File ~/mambaforge/envs/esmvaltool/lib/python3.10/site-packages/tropycal/tracks/dataset.py:204, in TrackDataset.__init__(self, basin, source, include_btk, interpolate_data, **kwargs)
    202     self.__read_hurdat()
    203 elif source == 'ibtracs':
--> 204     self.__read_ibtracs()
    205 else:
    206     raise RuntimeError("Accepted values for 'source' are 'hurdat' or 'ibtracs'")

File ~/mambaforge/envs/esmvaltool/lib/python3.10/site-packages/tropycal/tracks/dataset.py:678, in TrackDataset.__read_ibtracs(self)
    676 if ibtracs_online:
    677     path = self.ibtracs_url.replace("(basin)",ibtracs_basin)
--> 678     content = read_url(path)
    679 else:
    680     f = open(self.ibtracs_url,"r")

File ~/mambaforge/envs/esmvaltool/lib/python3.10/site-packages/tropycal/utils/generic_utils.py:1172, in read_url(url, split, subsplit)
   1170 def read_url(url,split=True,subsplit=True):
-> 1172     f = urllib.request.urlopen(url)
   1173     content = f.read()
   1174     content = content.decode("utf-8")

File ~/mambaforge/envs/esmvaltool/lib/python3.10/urllib/request.py:216, in urlopen(url, data, timeout, cafile, capath, cadefault, context)
    214 else:
    215     opener = _opener
--> 216 return opener.open(url, data, timeout)

File ~/mambaforge/envs/esmvaltool/lib/python3.10/urllib/request.py:525, in OpenerDirector.open(self, fullurl, data, timeout)
    523 for processor in self.process_response.get(protocol, []):
    524     meth = getattr(processor, meth_name)
--> 525     response = meth(req, response)
    527 return response

File ~/mambaforge/envs/esmvaltool/lib/python3.10/urllib/request.py:634, in HTTPErrorProcessor.http_response(self, request, response)
    631 # According to RFC 2616, "2xx" code indicates that the client's
    632 # request was successfully received, understood, and accepted.
    633 if not (200 <= code < 300):
--> 634     response = self.parent.error(
    635         'http', request, response, code, msg, hdrs)
    637 return response

File ~/mambaforge/envs/esmvaltool/lib/python3.10/urllib/request.py:563, in OpenerDirector.error(self, proto, *args)
    561 if http_err:
    562     args = (dict, 'default', 'http_error_default') + orig_args
--> 563     return self._call_chain(*args)

File ~/mambaforge/envs/esmvaltool/lib/python3.10/urllib/request.py:496, in OpenerDirector._call_chain(self, chain, kind, meth_name, *args)
    494 for handler in handlers:
    495     func = getattr(handler, meth_name)
--> 496     result = func(*args)
    497     if result is not None:
    498         return result

File ~/mambaforge/envs/esmvaltool/lib/python3.10/urllib/request.py:643, in HTTPDefaultErrorHandler.http_error_default(self, req, fp, code, msg, hdrs)
    642 def http_error_default(self, req, fp, code, msg, hdrs):
--> 643     raise HTTPError(req.full_url, code, msg, hdrs, fp)

HTTPError: HTTP Error 404: Not Found

Similar returns are for  `basin = tracks.TrackDataset(basin='australia',source='ibtracs')`.

By the way, I have updated the package to V5.

**Comment by tomerburg** [comment]
There are two things going on here:

1. There was indeed a bug with the South Atlantic data retrieval - part of the documentation incorrectly stated "south_america", which I now fixed to "south_atlantic" as well as adding a default map projection for plotting the South Atlantic basin. This fix also addresses your other open issue (#165) and will go into effect in the next version of Tropycal.

2. As far as the Australia projection, technically there is no Australia-only IBTrACS file, but you can access Australia-basin only data by retrieving a global IBTrACS TrackDataset object and retrieving Season objects while specifying the Australia basin:
- basin = tracks.TrackDataset(basin='all',source='ibtracs',ibtracs_url='ibtracs.csv')
- season = basin.get_season(2004,'australia')
- season.to_dataframe()

This is not necessarily a bug as Tropycal is providing data that isn't provided by default in IBTrACS, but a possible future feature is to add an option to partition the basin while declaring an instance of TrackDataset, as opposed to only through Season objects.

**Comment by tomerburg** [comment]
Update: I was able to add "australia" as an option to TrackDataset. Doing so retrieves the full IBTrACS dataset (which will take a minute or two if you don't have the file saved locally), then only retains storms in the Australian basin.

Running the following lines:

``basin = tracks.TrackDataset('australia','ibtracs')``
``basin.to_dataframe()``

Now outputs the following:

![image](https://user-images.githubusercontent.com/23174572/178576322-ee88dc38-a9d3-4acf-b690-4f02bd5cc025.png)

**Comment by GC-OCEAN** [comment]
Yes,
`basin = tracks.TrackDataset(basin='all',source='ibtracs')`
`season = basin.get_season(2004,'australia')`
works for me.

But,
`basin = tracks.TrackDataset('australia','ibtracs')` returns
<img width="850" alt="image" src="https://user-images.githubusercontent.com/50480172/178676172-f4e26eba-f888-45df-83e0-8a1a0f0726c9.png">

I am a little surprised that it worked for you. Could you tell me what this code does? The ALL file will be downloaded? Another concern is I don’t know what Australian basin refers to. What is the range of longitude and latitude?

**Comment by tomerburg** [comment]
I should specify these fixes I discussed are lined up for the v0.5.1 bug fix release, and are currently on the ``fix-bugs`` branch. Once this fix is merged into the ``master`` branch and released as a new Tropycal version, then running this code will download the ALL file and subset it to only include storms in the Australian basin (which is defined as 90E to 160E longitude south of the equator).
